scrapegraphai/graphs/base_graph.py

```diff
@@ -92,6 +93,9 @@ def _execute_standard(self, initial_state: dict) -> Tuple[dict, list]:
         """
         current_node_name = self.entry_point
         state = initial_state
+        self.shared_state["source_url"] = state["url"]
```

Comment:
- I don't like the handling of `shared_state`, seems a bit convoluted and difficult to maintain/extend in the future